api/base/serializers.py

```diff
@@ -770,7 +770,7 @@ def to_representation(self, value):
             raise ImproperlyConfigured(msg % self.view_name)
 
         if url is None:
-            raise SkipField
+            return {'data': None}
```

Comment:
I am wondering if an appropriate place to put this might be the `format_relationship_links` method.
```
def format_relationship_links(related_link=None, self_link=None, rel_meta=None, self_meta=None):
    """
    Properly handles formatting of self and related links according to JSON API.

    Removes related or self link, if none.
    """
    if not related_link and not self_link:
        return {'data': None}
  ...
```

You would still remove the SkipField and you would have to make a few changes to be passing in null related and self links where appropriate, such as `related_url = url['related'] if url else None`.   This will do the exact same thing as you're doing here, just containing the relationships logic in one place. Up to you!

Reply:
I see the need for clarity in this section of the code, but I'm hesitant to add more code then strictly necessary. I think there should be clean up around section, but I don't know if we can do easily in this case.

Reply:
It would just be about three lines of changes to existing code, but I'm pretty indifferent. I'll defer to you.